### PR TITLE
draft readme

### PR DESCRIPTION
**Pull request by robjweiss**
Here's a first draft of the readme.

Let me know if anything is incorrect.

I'd like to:
- Add a basic feature listing
- Add additional badges and have them link to their respective services
- Update the installation step once the package is available on NPM

### CODE REVIEW

**Comment by naggingant** [comment]
@robjweiss sorry this was my fault, fixed the links :+1:

File: README.md [hunk 1]
```diff
@@ -1,9 +1,86 @@
...
+```js
+const client = new Client(BROKER_URL, {
+    plugins: [
+        new plugins.GracefulShutdown(),
+        new plugins.ConnectionRetry(),
+        new plugins.Duplex(),
+        new plugins.Encoding('json'),
+        new plugins.RPC(),
+        new plugins.Confirm(),
+        new plugins.Retry()
```
Comment (naggingant) [inline README.md:39]: We could emphasize that users can use strings instead.
```node
    plugins: [
        'gracefulShutdown', // register with default options
        'connectionRetry',
        'duplex',
        new plugins.Encoding('json') // instantiate when you need a detailed configuration
    ]
```
But please note that this interface has the highest possibility of being modified in the near future.

Comment (naggingant) [reply README.md:39]: We should also note that one can just initialize without `plugins` option to start by default plugins.
`logger` option can be mentioned as well.

**Comment by robjweiss** [comment]
Thanks for the feedback. I tried to address some of those concerns and created two new issues:

- https://github.com/openrm/hato/issues/24
- https://github.com/openrm/hato/issues/25

**Pull request merged**